core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -183,7 +183,7 @@ object HostedPartition {
   /**
    * This broker hosts the partition, but it is in an offline log directory.
    */
-  final object Offline extends HostedPartition
+  final case class Offline(topicId: Option[Uuid]) extends HostedPartition
```

Comment:
Should this instead be `TopicIdPartition` as `Offline` here represent a partition and not a topic per-say?!

Reply:
Specially that the signature for `Online` is `final case class Online(partition: Partition)` so just to keep it some how consistent We can have `Offline(topicIdPartition: Option[TopicIdPartition]`

_Note_: this is just a suggestion!

Reply:
Addressed in [99dbbaa](https://github.com/apache/kafka/pull/15263/commits/99dbbaa9dbb90affbc56832b9ea2e546c27cb4d1). Used the `Partition` class for consistency. It has a method to retrieve topcicId.